api/src/server.ts

```diff
@@ -53,6 +53,10 @@ const start = async () => {
     genReqId: () => randomBytes(8).toString('hex'),
     disableRequestLogging: true
   });
+  fastify.addHook('onRequest', (req, _reply, done) => {
```

Comment:
This reduces the granular control on 'specific' requests that we may want to log - example: I always want the ping endpoint, but never the others unless the 'debug' filter is on.

I see your point about single source of truth -- but this makes it "too noisy" even when we turn on "debug" filters.

I would like more control.

Reply:
Okay, what would you like to see at a given log-level?

What I was thinking was that we'd want a certain baseline when we set the log-level to debug. i.e. what `requestSerializer` gives us: `method`, `url`, `ip`, `hostname`, `remoteAddress` and `remotePort`.  Then, if you want more for a specific route, you'd just do this:

```
\\ in route handler:

req.log.debug({ more: "stuff" }, "a useful message").
```

and see:

```
[16:26:28 UTC] DEBUG: incoming request {"reqId":"d8773cb47f268617","req":{"method":"GET","url":"/status/ping","ip":"127.0.0.1","hostname":"localhost:3000","remoteAddress":"127.0.0.1","remotePort":36760}}
[16:26:28 UTC] DEBUG: a useful message {"reqId":"d8773cb47f268617","more":"stuff"}
```

Reply:
The thing is this hook as is will log every request for every end-point. That is noisy. Sure we could update the baseline in the serializer, but we do not need to add this hook. 

I would prefer adding logs to some/few of our endpoints to begin with.

Reply:
I think we'll want to work out how to filter logs by endpoint and then we can turn them all on.

I do see your point, though, and I'll get rid of the hook.

Is there extra stuff you'd like to see in the serializer? If so, how about I update it and use it for this endpoint?

Reply:
> Is there extra stuff you'd like to see in the serializer? If so, how about I update it and use it for this endpoint?

Sure - go ahead and add anything you see fit. Consolidation is better. I do not have any specific needs other than the end user IP